src/promptflow-core/promptflow/contracts/flow.py

```diff
@@ -702,12 +702,14 @@ def _update_working_dir(cls, working_dir: Path):
         sys.path.insert(0, str(working_dir))
 
     @classmethod
-    def from_yaml(cls, flow_file: Path, working_dir=None) -> "Flow":
+    def from_yaml(cls, flow_file: Path, working_dir=None, name=None) -> "Flow":
         """Load flow from yaml file."""
         working_dir = cls._parse_working_dir(flow_file, working_dir)
         with open(working_dir / flow_file, "r", encoding=DEFAULT_ENCODING) as fin:
             flow_dag = load_yaml(fin)
-        flow_dag["name"] = flow_dag.get("name", _sanitize_python_variable_name(working_dir.stem))
+        # Name priority: name from yaml content > name from payload > working_dir.stem
```

Comment:
Maybe the name from payload should have the highest priority?

Reply:
Good catch, change the priority.